lib/blacklight/models/file.rb

```diff
@@ -11,11 +13,17 @@ class BlacklightFile
     def initialize(zip_entry)
       path = zip_entry.name
       id = File.basename(path)
+      xid = id[/__(xid-[0-9]+_[0-9]+)/, 1]
```

Comment:
I think [this](https://ruby-doc.org/core-2.2.0/String.html#method-i-5B-5D) is what you're looking for.